Rendering/WDE.MapSpawns/Rendering/SpawnContextMenu.cs

```diff
@@ -1,23 +1,111 @@
+using WDE.DatabaseEditors.Services;
+using WDE.DatabaseEditors.Data.Interfaces;
 
 namespace WDE.MapSpawns.Rendering;
 
 public class SpawnContextMenu
 {
     private readonly ISpawnSelectionService spawnSelectionService;
-    
+    private readonly IDatabaseQueryExecutor mySqlExecutor;
+    private readonly ITableDefinitionProvider definitionProvider;
+    private readonly IMessageBoxService messageBoxService;
+
     private ICommand CopyGuidCommand { get; }
+    private ICommand CopyPositionXCommand { get; }
+    private ICommand CopyPositionYCommand { get; }
+    private ICommand CopyPositionZCommand { get; }
+    private ICommand CopyOrientationCommand { get; }
+    private ICommand UpdateValuesCommand { get; }
 
-    public SpawnContextMenu(ISpawnSelectionService spawnSelectionService,
-        IClipboardService clipboardService)
+    public SpawnContextMenu(
+        ISpawnSelectionService spawnSelectionService,
+        IClipboardService clipboardService,
+        ITableEditorPickerService tableEditorPickerService,
+        IDatabaseQueryExecutor mySqlExecutor,
+        ITableDefinitionProvider definitionProvider,
+        IMessageBoxService messageBoxService)
     {
         this.spawnSelectionService = spawnSelectionService;
+        this.mySqlExecutor = mySqlExecutor;
+        this.definitionProvider = definitionProvider;
+        this.messageBoxService = messageBoxService;
 
         CopyGuidCommand = new DelegateCommand<SpawnInstance>(inst => clipboardService.SetText(inst.Guid.ToString()));
+        CopyPositionXCommand = new DelegateCommand<SpawnInstance>(inst => clipboardService.SetText(inst.WorldObject!.Position.X.ToString()));
+        CopyPositionYCommand = new DelegateCommand<SpawnInstance>(inst => clipboardService.SetText(inst.WorldObject!.Position.Y.ToString()));
+        CopyPositionZCommand = new DelegateCommand<SpawnInstance>(inst => clipboardService.SetText(inst.WorldObject!.Position.Z.ToString()));
+        CopyOrientationCommand = new DelegateCommand<SpawnInstance>(inst =>
+        {
+            CreatureSpawnInstance? creature = inst as CreatureSpawnInstance;
+            GameObjectSpawnInstance? go = inst as GameObjectSpawnInstance;
+
+            if (creature != null) 
+            { 
+                clipboardService.SetText(creature.Creature!.Orientation.ToString()); 
+            }
+            else if (go != null) 
+            {
+                clipboardService.SetText(go.GameObject!.Orientation.ToString());
+            }
+        });
+        UpdateValuesCommand = new DelegateCommand<SpawnInstance>(async (inst) =>
+        {
+            var trans = Queries.BeginTransaction();
+
+            CreatureSpawnInstance? creature = inst as CreatureSpawnInstance;
+            GameObjectSpawnInstance? go = inst as GameObjectSpawnInstance;
+
+            var typeString = creature != null ? "creature" : "gameobject";
+
+            trans.Table(typeString).InsertIgnore(new { Guid = inst.Guid });
```

Comment:
I think it is not a good idea to have an insert query here, this could insert an invalid row to the database (no creature entry, no phasemask, no spawnmask), beside you can only edit existing objects now, so probably better to remove it for now.